### PR TITLE
Added support of WB32 DFU for Windows.

### PR DESCRIPTION
**Pull request by wb-Joy**
Added support of WB32 DFU for Windows.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

*

### CODE REVIEW

**Comment by fauxpark** [comment]
Package should be submitted to MSYS2 and binary taken from there.
https://github.com/msys2/mingw-packages
http://packages.msys2.org/

**Comment by wb-Joy** [comment]
@fauxpark The executable has been uploaded to mingw-Packages.
https://packages.msys2.org/base/mingw-w64-wb32-dfu-updater

**Comment by fauxpark** [comment]
@wb-Joy these two PRs need to be updated per #330 and #331.

**Comment by wb-Joy** [comment]
> @wb-Joy these two PRs need to be updated per #330 and #331.

I have updated.

File: windows/QMK Toolbox/Usb/Bootloader/BootloaderType.cs [hunk 1]
```diff
@@ -14,6 +14,7 @@ public enum BootloaderType
         QmkDfu,
         Stm32Dfu,
         Stm32Duino,
+        Wb32Dfu,
         UsbAsp,
         UsbTinyIsp,
```
Comment (fauxpark) [inline windows/QMK Toolbox/Usb/Bootloader/BootloaderType.cs:19]:
```suggestion
        UsbAsp,
        UsbTinyIsp,
        Wb32Dfu,
```

File: windows/QMK Toolbox/QMK Toolbox.csproj [hunk 2]
```diff
@@ -187,6 +188,7 @@
     <EmbeddedResource Include="Resources\dfu-util.exe" />
     <EmbeddedResource Include="Resources\mdloader.exe" />
     <EmbeddedResource Include="Resources\qmk_driver_installer.exe" />
+    <EmbeddedResource Include="Resources\wb32-dfu-updater_cli.exe" />
     <EmbeddedResource Include="Resources\teensy_loader_cli.exe" />
```
Comment (fauxpark) [inline windows/QMK Toolbox/QMK Toolbox.csproj:192]:
```suggestion
    <EmbeddedResource Include="Resources\teensy_loader_cli.exe" />
    <EmbeddedResource Include="Resources\wb32-dfu-updater_cli.exe" />
```

File: windows/QMK Toolbox/Usb/UsbListener.cs [hunk 3]
```diff
@@ -217,6 +219,12 @@ private static BootloaderType GetDeviceType(ushort vendorId, ushort productId, u
                         return BootloaderType.Stm32Dfu;
                     }
                     break;
+                case 0x342D: // WestBerryTech
```
Comment (fauxpark) [inline windows/QMK Toolbox/Usb/UsbListener.cs:222]: Move below `0x314B`

File: windows/QMK Toolbox/Usb/Bootloader/Wb32DfuDevice.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            {
+                await RunProcessAsync("wb32-dfu-updater_cli.exe", $"--toolbox-mode --dfuse-address 0x08000000 --download \"{file}\"");
+            }
+            else if (Path.GetExtension(file)?.ToLower() == ".hex")
+            {
+                await RunProcessAsync("wb32-dfu-updater_cli.exe", $"--toolbox-mode --download \"{file}\"");
+            }
+            else
+            {
+                PrintMessage("This file format is not the type supported by wB32-dFU-updater_cli!", MessageType.Error);
```
Comment (fauxpark) [inline windows/QMK Toolbox/Usb/Bootloader/Wb32DfuDevice.cs:28]:
```suggestion
                PrintMessage("Only firmware files in .bin or .hex format can be flashed with wb32-dfu-updater_cli!", MessageType.Error);
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -56,7 +56,7 @@ jobs:
     - name: Check binary permissions
       working-directory: ./macos/QMK Toolbox/Resources
       env:
-        QMK_BINARIES: avrdude bootloadHID dfu-programmer dfu-util mdloader teensy_loader_cli
+        QMK_BINARIES: avrdude bootloadHID dfu-programmer dfu-util mdloader teensy_loader_cli wb32-dfu-updater_cli
```
Comment (fauxpark) [inline .github/workflows/build.yml:59]: Not needed for Windows, please put this in the macOS PR.

Comment (wb-Joy) [reply .github/workflows/build.yml:59]: ok

**Comment by fauxpark** [comment]
Files do not match.

This PR:
```
fauxpark@marissa:downloads$ md5 wb32-dfu-updater_cli.exe 
MD5 (wb32-dfu-updater_cli.exe) = 3be6125560a58e6d8c1efd13eb241ae1
```

[mingw-w64-x86_64-wb32-dfu-updater](https://packages.msys2.org/package/mingw-w64-x86_64-wb32-dfu-updater?repo=mingw64):
```
fauxpark@marissa:mingw-w64-x86_64-wb32-dfu-updater-1.0.0-1-any.pkg$ md5 mingw64/bin/wb32-dfu-updater_cli.exe
MD5 (mingw64/bin/wb32-dfu-updater_cli.exe) = d9af6cd88011a9fa9998e89e44ae82a7
```

**Comment by wb-Joy** [comment]
> Files do not match.
> 
> This PR:
> 
> ```
> fauxpark@marissa:downloads$ md5 wb32-dfu-updater_cli.exe 
> MD5 (wb32-dfu-updater_cli.exe) = 3be6125560a58e6d8c1efd13eb241ae1
> ```
> 
> [mingw-w64-x86_64-wb32-dfu-updater](https://packages.msys2.org/package/mingw-w64-x86_64-wb32-dfu-updater?repo=mingw64):
> 
> ```
> fauxpark@marissa:mingw-w64-x86_64-wb32-dfu-updater-1.0.0-1-any.pkg$ md5 mingw64/bin/wb32-dfu-updater_cli.exe
> MD5 (mingw64/bin/wb32-dfu-updater_cli.exe) = d9af6cd88011a9fa9998e89e44ae82a7
> ```

Sorry, I forgot to update to the latest version. It has now been updated.

**Comment by wb-Joy** [comment]
Can the Windows version be incorporated first?

**Comment by wb-Joy** [comment]
Is there anything else that needs to be changed? not merged.

**Review by fauxpark — APPROVED** [review]

**Pull request merged**